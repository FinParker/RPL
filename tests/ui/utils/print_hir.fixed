#![feature(stmt_expr_attributes)]

 //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
use std::sync::Arc; //~ NOTE: use std::sync::Arc;

 //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
mod m {
    //~^ NOTE: mod m {
    pub fn foo() {}
}

 //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
trait Foo {
    //~^ NOTE: trait Foo {
     //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    const N: usize; //~ NOTE: const N: usize;
}

 //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
impl Foo for () {
    //~^ NOTE: impl Foo for () {
     //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    const N: usize = 0_usize; //~ NOTE: const N: usize = 0usize;
}

 //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
fn main() {
    //~^ NOTE: fn main() {
     //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    let x = Arc::new(0_usize); //~ NOTE: let x = Arc::new(0usize);

     //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    fn foo() {
        //~^ NOTE: fn foo() {
         //~ ERROR: abort due to debugging
        //~^ HELP: remove this attribute
        //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
        //~| NOTE: this error is to remind you removing these attributes
        {} //~ NOTE: { }
    }

     //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    if true {
        //~^ NOTE: if true {
    } else {
    }

     //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    std::thread::spawn(move || {
        //~^ NOTE: std::thread::spawn(move ||
        println!("{x}");
    });

     //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    macro_rules! mac {
        //~^ NOTE: macro_rules! mac {
        () => {
            #[rpl::print_hir] // No effect after expansion.
            println!("test");
        };
    }

    #[rpl::print_hir] // No effect after expansion.
    mac!();
}
