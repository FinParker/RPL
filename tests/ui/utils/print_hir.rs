#![feature(stmt_expr_attributes)]

#[rpl::print_hir] //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
use std::sync::Arc; //~ NOTE: use std::sync::Arc;

#[rpl::print_hir] //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
mod m {
    //~^ NOTE: mod m {
    pub fn foo() {}
}

#[rpl::print_hir] //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
trait Foo {
    //~^ NOTE: trait Foo {
    #[rpl::print_hir] //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    const N: usize; //~ NOTE: const N: usize;
}

#[rpl::print_hir] //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
impl Foo for () {
    //~^ NOTE: impl Foo for () {
    #[rpl::print_hir] //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    const N: usize = 0_usize; //~ NOTE: const N: usize = 0usize;
}

#[rpl::print_hir] //~ ERROR: abort due to debugging
//~^ HELP: remove this attribute
//~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
//~| NOTE: this error is to remind you removing these attributes
fn main() {
    //~^ NOTE: fn main() {
    #[rpl::print_hir] //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    let x = Arc::new(0_usize); //~ NOTE: let x = Arc::new(0usize);

    #[rpl::print_hir] //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    fn foo() {
        //~^ NOTE: fn foo() {
        #[rpl::print_hir] //~ ERROR: abort due to debugging
        //~^ HELP: remove this attribute
        //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
        //~| NOTE: this error is to remind you removing these attributes
        {} //~ NOTE: { }
    }

    #[rpl::print_hir] //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    if true {
        //~^ NOTE: if true {
    } else {
    }

    #[rpl::print_hir] //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    std::thread::spawn(move || {
        //~^ NOTE: std::thread::spawn(move ||
        println!("{x}");
    });

    #[rpl::print_hir] //~ ERROR: abort due to debugging
    //~^ HELP: remove this attribute
    //~| NOTE: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
    //~| NOTE: this error is to remind you removing these attributes
    macro_rules! mac {
        //~^ NOTE: macro_rules! mac {
        () => {
            #[rpl::print_hir] // No effect after expansion.
            println!("test");
        };
    }

    #[rpl::print_hir] // No effect after expansion.
    mac!();
}
