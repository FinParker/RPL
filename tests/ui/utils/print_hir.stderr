note: HirId(DefId(0:3 ~ print_hir[654a]::{use#1}).0) (`use std::sync::Arc;`)
  --> tests/ui/utils/print_hir.rs:7:5
   |
LL | #[rpl::print_hir]
   | ----------------- HIR printed because of this attribute
...
LL | use std::sync::Arc;
   |     ^^^^^^^^^^^^^^

note: HirId(DefId(0:4 ~ print_hir[654a]::m).0) (`mod m {
          fn foo() { }
      }`)
  --> tests/ui/utils/print_hir.rs:13:1
   |
LL | #[rpl::print_hir]
   | ----------------- HIR printed because of this attribute
...
LL | mod m {
   | ^^^^^

note: HirId(DefId(0:6 ~ print_hir[654a]::Foo).0) (`trait Foo {
          const
          N:
          usize;
      }`)
  --> tests/ui/utils/print_hir.rs:22:1
   |
LL | #[rpl::print_hir]
   | ----------------- HIR printed because of this attribute
...
LL | trait Foo {
   | ^^^^^^^^^

note: HirId(DefId(0:7 ~ print_hir[654a]::Foo::N).0) (`const N: usize;`)
  --> tests/ui/utils/print_hir.rs:28:5
   |
LL |     #[rpl::print_hir]
   |     ----------------- HIR printed because of this attribute
...
LL |     const N: usize;
   |     ^^^^^^^^^^^^^^

note: HirId(DefId(0:8 ~ print_hir[654a]::{impl#0}).0) (`impl Foo for () {
          const
          N:
          usize
          =
          0usize;
      }`)
  --> tests/ui/utils/print_hir.rs:35:1
   |
LL | #[rpl::print_hir]
   | ----------------- HIR printed because of this attribute
...
LL | impl Foo for () {
   | ^^^^^^^^^^^^^^^

note: HirId(DefId(0:9 ~ print_hir[654a]::{impl#0}::N).0) (`const N: usize = 0usize;`)
  --> tests/ui/utils/print_hir.rs:41:5
   |
LL |     #[rpl::print_hir]
   |     ----------------- HIR printed because of this attribute
...
LL |     const N: usize = 0_usize;
   |     ^^^^^^^^^^^^^^

note: HirId(DefId(0:10 ~ print_hir[654a]::main).0) (`fn main() {
          let x = Arc::new(0usize);
          fn foo() { { } }
          if true { } else { }
                  std::thread::spawn(move ||
                          {
                              {
                                  ::std::io::_print(format_arguments::new_v1(&["", "\n"],
                                          &[format_argument::new_display(&x)]));
                              };
                          });
                  macro_rules! mac {
                      () => { #[rpl::print_hir] println!("test"); };
                  }
                  { ::std::io::_print(format_arguments::new_const(&["test\n"])); };
              }`)
  --> tests/ui/utils/print_hir.rs:48:1
   |
LL | #[rpl::print_hir]
   | ----------------- HIR printed because of this attribute
...
LL | fn main() {
   | ^^^^^^^^^

note: HirId(DefId(0:10 ~ print_hir[654a]::main).2) (`let x = Arc::new(0usize);`)
  --> tests/ui/utils/print_hir.rs:54:5
   |
LL |     #[rpl::print_hir]
   |     ----------------- HIR printed because of this attribute
...
LL |     let x = Arc::new(0_usize);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

note: HirId(DefId(0:11 ~ print_hir[654a]::main::foo).0) (`fn foo() { { } }`)
  --> tests/ui/utils/print_hir.rs:60:5
   |
LL |     #[rpl::print_hir]
   |     ----------------- HIR printed because of this attribute
...
LL |     fn foo() {
   |     ^^^^^^^^

note: HirId(DefId(0:11 ~ print_hir[654a]::main::foo).2) (`{ }`)
  --> tests/ui/utils/print_hir.rs:66:9
   |
LL |         #[rpl::print_hir]
   |         ----------------- HIR printed because of this attribute
...
LL |         {}
   |         ^^

note: HirId(DefId(0:10 ~ print_hir[654a]::main).12) (expr `if true {

          } else {
          }`)
  --> tests/ui/utils/print_hir.rs:73:5
   |
LL |       #[rpl::print_hir]
   |       ----------------- HIR printed because of this attribute
...
LL | /     if true {
LL | |
LL | |     } else {
LL | |     }
   | |_____^

note: HirId(DefId(0:10 ~ print_hir[654a]::main).20) (`std::thread::spawn(move ||
              {
                  {
                      ::std::io::_print(format_arguments::new_v1(&["", "\n"],
                              &[format_argument::new_display(&x)]));
                  };
              })`)
  --> tests/ui/utils/print_hir.rs:82:5
   |
LL |       #[rpl::print_hir]
   |       ----------------- HIR printed because of this attribute
...
LL | /     std::thread::spawn(move || {
LL | |
LL | |         println!("{x}");
LL | |     });
   | |______^

note: HirId(DefId(0:13 ~ print_hir[654a]::main::mac).0) (`macro_rules! mac { () => { #[rpl::print_hir] println!("test"); }; }`)
  --> tests/ui/utils/print_hir.rs:91:5
   |
LL |     #[rpl::print_hir]
   |     ----------------- HIR printed because of this attribute
...
LL |     macro_rules! mac {
   |     ^^^^^^^^^^^^^^^^

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:3:1
   |
LL | #[rpl::print_hir]
   | ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:9:1
   |
LL | #[rpl::print_hir]
   | ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:18:1
   |
LL | #[rpl::print_hir]
   | ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:24:5
   |
LL |     #[rpl::print_hir]
   |     ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:31:1
   |
LL | #[rpl::print_hir]
   | ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:37:5
   |
LL |     #[rpl::print_hir]
   |     ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:44:1
   |
LL | #[rpl::print_hir]
   | ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:50:5
   |
LL |     #[rpl::print_hir]
   |     ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:56:5
   |
LL |     #[rpl::print_hir]
   |     ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:62:9
   |
LL |         #[rpl::print_hir]
   |         ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:69:5
   |
LL |     #[rpl::print_hir]
   |     ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:78:5
   |
LL |     #[rpl::print_hir]
   |     ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: abort due to debugging
  --> tests/ui/utils/print_hir.rs:87:5
   |
LL |     #[rpl::print_hir]
   |     ^^^^^^^^^^^^^^^^^ help: remove this attribute
   |
   = note: `#[rpl::dump_hir]`, `#[rpl::print_hir]` and `#[rpl::dump_mir]` are only used for debugging
   = note: this error is to remind you removing these attributes

error: aborting due to 13 previous errors

